tests/unattended/unit-tests.sh

```diff
@@ -0,0 +1,197 @@
+trap clean SIGINT
+
+logfile="./unit-tests.log"
+echo "-------------------------" >> ./unit-tests.log
+debug=">> ${logfile}"
+ALL_FILES=("common" "checks")
+
+function logger() {
+
+    now=$(date +'%d/%m/%Y %H:%M:%S')
+    case ${1} in 
+        "-e")
+            mtype="ERROR:"
+            message="${2}"
+            ;;
+        "-w")
+            mtype="WARNING:"
+            message="${2}"
+            ;;
+        *)
+            mtype="INFO:"
+            message="${1}"
+            ;;
+    esac
+    echo "${now} ${mtype} ${message}" | tee -a ${logfile}
+
+}
+
+
+function createImage() {
+
+    if [ ! -f ./Dockerfile ]; then
+        logger -e "No Dockerfile found to create the environment."
+        exit 1
+    fi
+
+    image_name="testing-img"
+    if [ -z "$(docker images | grep $image_name)" ]; then
+        eval "docker build -t $image_name . ${debug}"
+        if [ "$?" != 0 ]; then
+            logger -e "Docker encountered some error."
+            exit 1
+        else 
+            logger "Docker image created successfully."
+        fi
+    else 
+        logger "Docker image found."
+    fi
+}
+
+function runContainer() {
+    container_name="testing-container"
+    eval "docker run -d -t --name $container_name $image_name /bin/bash ${debug}"
+    if [ "$?" != 0 ]; then
+        logger -e "Docker encountered some error."
+        exit 1
+    else 
+        logger "Docker container created successfully."
+    fi
+    container_id="$( docker ps -a | grep $container_name | awk '{ print $1 }' )"
+    eval "docker cp bach.sh $container_id:/tests/unattended/bach.sh ${debug}"
+    if [ "$?" != 0 ]; then
+        logger -e "Error copying bach.sh to the container."
+        exit 1
+    fi
+}
+
+function testFile() {
+
+    logger "Unit tests for $1.sh."
+
+    eval "docker cp test-$1.sh $container_id:/tests/unattended/test-$1.sh ${debug}"
+    if [ "$?" != 0 ]; then
+        logger -e "File test-$1.sh could not be copied to the container."
+        return
+    fi
+    eval "mkdir -p temp/ ${debug}"
+
+    if [ -f ../../unattended_installer/install_functions/opendistro/$1.sh ]; then
+        eval "cp ../../unattended_installer/install_functions/opendistro/$1.sh temp/ ${debug}"
+    elif [ -f ../../unattended_installer/install_functions/elasticsearch_basic/$1.sh ]; then
+        eval "cp ../../unattended_installer/install_functions/elasticsearch_basic/$1.sh temp/ ${debug}"
+    elif [ -f ../../unattended_installer/$1.sh ]; then
+        eval "cp ../../unattended_installer/$1.sh ${debug}"
+    else 
+        logger -e "File $1.sh could not be found."
+        return
+    fi
+    
+    eval "docker cp temp/$1.sh $container_id:/tests/unattended/$1.sh ${debug}"
+    if [ "$?" != 0 ]; then
+        logger -e "File $1.sh could not be copied to the container."
+        return
+    fi
+
+    eval "docker exec -it $container_name env TERM=xterm-256color bash -c \"cd /tests/unattended && bash test-$1.sh\" | tee -a ${logfile}"
```

Comment:
This must be defined in an Entrypoint

Reply:
Added in https://github.com/wazuh/wazuh-packages/pull/1136/commits/ed078bbe9b25f9f52ebe0076b232fc364ff2edc1